scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/TaskResultCreator.java

```diff
@@ -0,0 +1,143 @@
+ * the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s): ActiveEon Team - http://www.activeeon.com
+ *
+ *  * $$ACTIVEEON_CONTRIBUTOR$$
+ */
+package org.ow2.proactive.scheduler.core.helpers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.ow2.proactive.db.DatabaseManagerException;
+import org.ow2.proactive.scheduler.common.exception.UnknownTaskException;
+import org.ow2.proactive.scheduler.common.job.JobType;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.TaskLogs;
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.core.db.SchedulerDBManager;
+import org.ow2.proactive.scheduler.descriptor.EligibleTaskDescriptor;
+import org.ow2.proactive.scheduler.descriptor.JobDescriptor;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.task.TaskResultImpl;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+
+
+public class TaskResultCreator {
+
+    private static TaskResultCreator instance = null;
+
+    private TaskResultCreator() {
+    }
+
+    public static synchronized TaskResultCreator getInstance() {
+        if (instance == null) {
+            instance = new TaskResultCreator();
+        }
+        return instance;
+    }
+
+    public TaskResultImpl getTaskResult(SchedulerDBManager dbManager, InternalJob job, InternalTask task) 
+            throws UnknownTaskException  {
+        return getTaskResult(dbManager, job, task, null, null);
+    }
+
+    public TaskResultImpl getTaskResult(SchedulerDBManager dbManager, InternalJob job, InternalTask task,
+            Throwable exception, TaskLogs output) throws UnknownTaskException {
+        if (task == null){
+            throw new UnknownTaskException();
+        }
+
+        TaskResultImpl taskResult;
+        JobDescriptor jobDescriptor = job.getJobDescriptor();
+        try{
+            taskResult = (TaskResultImpl) dbManager.loadTasksResults(job.getId(), 
+                    Collections.singletonList(task.getId())).get(task.getId());
+        }catch(DatabaseManagerException e){
```

Comment:
Can we avoid this exception somehow? (http://c2.com/cgi/wiki?DontUseExceptionsForFlowControl)